src/main/scala/fr/acinq/bitcoin/Crypto.scala

```diff
@@ -147,96 +133,90 @@ object Crypto {
+    }
 
-    // used only if secp256k1 is not available
-    lazy val ecpoint = curve.getCurve.decodePoint(value.toArray)
-    /**
-      *
-      * @return a binary representation of this point in DER format
-      */
-    def toBin(compressed: Boolean): ByteVector = compressed match {
-      case false => decompress(value)
-      case true => compress(value)
+    def multiply(that: PrivateKey): PublicKey = if (Secp256k1Context.isEnabled) {
+      PublicKey.deserialize(ByteVector.view(NativeSecp256k1.pubKeyTweakMul(value.toArray, that.toBin.toArray)))
+    } else {
+      PublicKey(ecpoint.multiply(that.bigInt).normalize())
     }
 
-    // because ECPoint is not serializable
-    protected def writeReplace: Object = PointProxy(toBin(false))
+    def +(that: PublicKey): PublicKey = add(that)
 
-    override def toString = toBin(true).toHex
+    def -(that: PublicKey): PublicKey = substract(that)
 
-    override def hashCode(): Int = toBin(true).hashCode
+    def *(that: PrivateKey): PublicKey = multiply(that)
 
-    override def equals(obj: Any): Boolean = obj match {
-      case null => false
-      case p: Point => toBin(true) == p.toBin(true)
-      case _ => false
-    }
-  }
+    def toBin: ByteVector = value
 
-  case class PointProxy(bin: ByteVector) {
-    def readResolve: Object = Point(bin)
-  }
+    def serialize(compressed: Boolean): ByteVector = compressed match {
+      case true => value
+      case false if Secp256k1Context.isEnabled => ByteVector.view(NativeSecp256k1.parsePubkey(value.toArray))
+      case false => ByteVector.view(ecpoint.getEncoded(false))
+    }
 
-  object Point {
-    def apply(data: ECPoint): Point = new Point(ByteVector.view(data.getEncoded(false)))
+    override def toString = toBin.toHex
 
-    def isValid(point: ByteVector): Boolean = isPubKeyValid(point) && {
-      if (Secp256k1Context.isEnabled)
-        NativeSecp256k1.parsePubkey(point.toArray).length == 65
-      else
-        curve.getCurve.decodePoint(point.toArray).normalize().isValid
-    }
+    // used only if secp256k1 is not available
+    lazy val ecpoint = curve.getCurve.decodePoint(value.toArray)
   }
 
   object PublicKey {
+    def apply(data: ECPoint): PublicKey = new PublicKey(ByteVector.view(data.getEncoded(true)))
 
     /**
       * @param raw        serialized value of this public key (a point)
       * @param checkValid indicates whether or not we check that this is a valid public key; this should be used
       *                   carefully for optimization purposes
       * @return
       */
-    def apply(raw: ByteVector, checkValid: Boolean = true): PublicKey = {
-      val pub = new PublicKey(raw)
-      if (checkValid) {
-        // this is expensive and done only if needed
-        require(Point.isValid(raw))
-      }
-      pub
+    def apply(raw: ByteVector, checkValid: Boolean): PublicKey = deserialize(raw, checkValid)
+
+    def isValid(pub: ByteVector): Boolean = isPubKeyValid(pub) && {
```

Comment:
```suggestion
    def isValidStrict(pub: ByteVector): Boolean = isPubKeyValid(pub) && {
```

`isValid` and `isValidStrict` should be defined next to each other